src/net/RTP/AudioStream.cs

```diff
@@ -100,29 +106,43 @@ public void SendAudioFrame(uint duration, int payloadTypeID, byte[] buffer)
                     // paylaod.
                     // See https://github.com/sipsorcery/sipsorcery/issues/394.
 
-                    uint payloadDuration = 0;
+                    int maxPayload = RTPSession.RTP_MAX_PAYLOAD;
+                    int totalPackets = (buffer.Length + maxPayload - 1) / maxPayload;
+
+                    uint totalIncrement = 0;
+                    uint startTimestamp = LocalTrack.Timestamp; // Keep track of where we started.
```

Comment:
Is there a reason for this change? Is it just doing the same thing as the original code but in a different way?

Reply:
yeah the original method has bugs.
In the for loop you have 
LocalTrack.Timestamp += payloadDuration;
and then on exit you have
 LocalTrack.Timestamp += duration;
 
 This works if the buffer doesn't exceed RTP_MAX_PAYLOAD but if it does it's adding on the duration twice.
 Also
 payloadDuration = (uint)(((decimal)payloadLength / buffer.Length) * duration);
 
 has rounding errors and results in drift